packages/dev/query/src/cli.js

```diff
@@ -569,32 +601,64 @@ export function run(input: string[]) {
       asset_node_modules: 0,
       asset_source: 0,
     };
+
+    let bg_ext = {};
+
+    let bg_type = {
+      entries: 0,
+      shared: 0,
+      async: 0,
+      parallel: 0,
+    };
+
     for (let [, n] of bundleGraph._graph.nodes) {
-      if (n.type in bg) {
+      if (n.type == 'bundle_group') {
+        // TODO: Counting priority here seems to work instead of counting on BundleNodes.. Counting DependencyNode priorities themselves
+        // gives different counts as well.
+        let priority = _getBundlePriority(bundleGraph, n);
+        if (priority == Priority.lazy) {
+          bg_type.async++;
+        } else if (priority == Priority.parallel) {
+          bg_type.parallel++;
+        }
+      } else if (n.type == 'bundle') {
+        bg.bundle++;
+
         // $FlowFixMe
-        bg[n.type]++;
-      }
-      if (n.type === 'asset') {
+        bg_ext[n.value.type] = (bg_ext[n.value.type] || 0) + 1;
+
+        // In general, !bundle.mainEntryId means that it is shared. Ignores async/shared edge cases.
+        // $FlowFixMe
+        if (n.value.mainEntryId == null) {
+          bg_type.shared++;
+        }
+      } else if (n.type == 'asset') {
         if (
+          // $FlowFixMe
           fromProjectPathRelative(n.value.filePath).includes('node_modules')
         ) {
           bg.asset_node_modules++;
         } else {
           bg.asset_source++;
         }
+      } else if (n.type == 'dependency') {
+        bg.dependency++;
+      } else if (n.type == 'entry_file') {
+        bg_type.entries++;
       }
     }
 
-    console.log('# Asset Graph Node Counts');
-    for (let k in ag) {
-      console.log(k, ag[k]);
+    let sum = 0;
+    for (let k in bg_ext) {
+      sum += bg_ext[k];
     }
-    console.log();
 
-    console.log('# Bundle Graph Node Counts');
-    for (let k in bg) {
-      console.log(k, bg[k]);
-    }
+    invariant(bg.bundle == sum);
+
+    _printStatsTable('# Asset Graph Node Counts', Object.entries(ag));
+    _printStatsTable('# Bundle Graph Node Counts', Object.entries(bg));
```

Comment:
I notice the BundleGraph table has `asset` count as well as `asset_nodemodules` and `asset_source` I think we can either remove the `asset` row for the bundleGraph table since we already have that count in the `AssetGraph` table, or, you can sum `asset_nodemodules + asset_source` to get the full count. Currently your example said 0 assets but the BundleGraph should contain all the asset nodes present in the AssetGraph

Reply:
Removing `asset` would make more sense to avoid being redundant